engine/core/README.md

```diff
@@ -0,0 +1,95 @@
+## Detailed implementation
+
+The config resolution algorithm is based on two basic building blocks: **Resolvers** and **Resolvables**, when put together they form the **Configuration State**.
+
+### Resolvers
+
+Resolvers are functions that has an arbitrary name, receives a context and an input, and produces an output, resolvers can be either sync or async and can also optionally returns a resolvable. The default resolver receives an input and produces the same input as received.
+
+```ts
+export type AsyncResolver<
+  T = any,
+  TParent = any,
+  TContext extends BaseContext = BaseContext
+> = (parent: TParent, context: TContext) => Promise<Resolvable<T> | T>;
+
+export type SyncResolver<
+  T = any,
+  TParent = any,
+  TContext extends BaseContext = BaseContext
+> = (parent: TParent, context: TContext) => Resolvable<T> | T;
+
+export type Resolver<
+  T = any,
+  TParent = any,
+  TContext extends BaseContext = BaseContext
+> = AsyncResolver<T, TParent, TContext> | SyncResolver<T, TParent, TContext>;
+```
+
+Meaning that a `Resolver<T, TInput>` is whatever function that takes TInput as a parameter and returns T or Resolvable<T> as an output.
+
+### Resolvable
+
+**Resolvables** are arbitrary json that contains a property named `__resolveType` which points to either: A resolver or a resolvable. When pointing to a **Resolver** the data is passed as a input to such resolver and when pointing to a **Resolvable** the algorithm starts again taking the target resolvable as input for the `__resolveType` property.
+
+When the reference for `__resolveType` is missing, a `Dangling reference` error is thrown to sinalize that the configuration state is invalid and must be fixed.
```

Comment:
Wdyt about adding an example with this dangling reference? You can move this to after the example below and add an example

Reply:
Done